sklearn/ensemble/tests/test_weight_boosting.py

```diff
@@ -486,3 +485,49 @@ def predict(self, X):
     boost = AdaBoostRegressor(DummyEstimator(), n_estimators=3)
     boost.fit(X, y_regr)
     assert_equal(len(boost.estimator_weights_), len(boost.estimator_errors_))
+
+
+def test_multidimensional_X():
+    """
+    Check that the AdaBoost estimators can work with n-dimensional
+    data matrix
+    """
+    class DummyClassifier(BaseEstimator):
+
+        def fit(self, X, y, sample_weight=None):
+            self.classes_ = np.unique(y)
+            return self
+
+        def predict(self, X):
+            n_samples = len(X)
+            predictions = np.random.choice(self.classes_, n_samples)
+            return predictions
+
+        def predict_proba(self, X):
+            n_samples = len(X)
+            n_classes = len(self.classes_)
+            probas = np.random.randn(n_samples, n_classes)
+            return probas
+
+    class DummyRegressor(BaseEstimator):
+
+        def fit(self, X, y):
+            return self
+
+        def predict(self, X):
+            n_samples = len(X)
+            predictions = np.random.randn(n_samples)
+            return predictions
+
+    X = np.random.randn(50, 3, 3)
+    yc = np.random.choice([0, 1], 50)
+    yr = np.random.randn(50)
+
+    boost = AdaBoostClassifier(DummyClassifier())
```

Comment:
why not use sklearn.dummy.DummyClassifier ?

also avoid using directly np.random and prefer using a RandomState